# Real Robot configuration
gen3_ip: "192.168.151.130"

# simulation Robot configuration
is_simulation: true
# where?
mujoco_xml_path: "/home/taehoyun/workspace/lee_backup/ros2_ws/src/robot_model/kinova_gen3/mujoco_model/kinova_gen3.xml"
# mujoco_xml_path: "/home/lee8646/ros2_ws/src/robot_model/kinova_gen3/mujoco_model/kinova_gen3.xml"
# mujoco_xml_path: "/home/wookie/ros2_ws/src/robot_model/kinova_gen3/mujoco_model/kinova_gen3_flex.xml"
#mujoco_xml_path: "/home/wookie/ros2_ws/src/robot_model/kinova_gen3/mujoco_model/kinova_gen3_2f85_reduced.xml"
simulation:
  joint_axis: [2,2,2,2,2,2,2]
  # theta_init: [0, -0.2618, 0.0, 1.833, 0, 1.571, 0] # for simulation configuration
  theta_init: [0, 0.386, 0.0, 2.29, 0, -1.07, 0] # for real robot configuration
  # theta_init: [0, 0, 0.0, 0, 0, 0, 0]

# Common Robot Configuration
# pinocchio_urdf_path: "/home/nuc/seowook/ros2_ws/src/robot_model/kinova_gen3/pinocchio_model/pinocchio_urdf/kinova_conti.urdf"
pinocchio_urdf_path: "/home/taehoyun/workspace/lee_backup/ros2_ws/src/robot_model/kinova_gen3/pinocchio_model/pinocchio_urdf/kinova_conti.urdf"

# pinocchio_urdf_path: "/home/lee8646/ros2_ws/src/robot_model/kinova_gen3/pinocchio_model/pinocchio_urdf/kinova_conti.urdf"
#pinocchio_urdf_path: "/home/wookie/ros2_ws/src/robot_model/kinova_gen3/pinocchio_model/pinocchio_urdf/kinova_conti_2f85.urdf"
lpf_tau_j_cutoff_freq: 100.0 #[Hz]
lpf_dq_cutoff_freq: 500.0 #[Hz]
is_rigid: true
dt: 0.001 # 1ms controller update rate

# Real F/T sensor configuration
ft_sensor_ip: "192.168.50.177"
base_ft: false
base_ft_frame_names: ["base_ft_sensor_joint"] # F/T sensor joint name in urdf
base_link_names: ["base_fixed"] # before first actuated joint name in urdf
base_joint_names: ["Actuator1"] # joint name attached to base 

# world | sensor joint | sensor mount | base link | Actuator1

# for optimal control
lpf_opt_u_cutoff_freq: 10.0

################################
### Controller Configuration ###
################################
init_controller: "GRAVITY_COMPENSATION" #JOINT_NRIC, JOINT_PD, TASK_NRIC, TASK_PD, GRAVITY_COMPENSATION, JOINT_PD_FRIC, TASK_PD_FRIC
joint_stiffness: [8000,8000,8000,8000,7000,7000,7000]
rotor_inertia: [0.4, 0.4, 0.4, 0.4, 0.2, 0.2, 0.2]
is_controller_logging: true

########################
### Controller Gains ###
########################
# JOINT_PD: u = Kp(qd - q) + Kd(qd_dot - q_dot)
joint_pd_gains:
  Kp: [100.0, 100.0, 100.0, 100.0, 70.0, 70.0, 70.0]
  Kd: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
  verbose: false

joint_nric_gains:
  k1: [8.0, 8.0, 8.0, 8.0, 5.0, 5.0, 5.0] # d-gian
  k2: [21.0, 21.0, 21.0, 21.0, 13.0, 13.0, 13.0] # p-gain
  K: [30.0, 30.0, 30.0, 30.0, 20.0, 20.0, 20.0]
  Kp: [13.0, 13.0, 13.0, 13.0, 8.0, 8.0, 8.0]
  Ki: [4.0, 4.0, 4.0, 4.0, 2.0, 2.0, 2.0]
  gamma: [4.0, 4.0, 4.0, 4.0, 4.0, 4.0, 4.0]
  reflected_inertia: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
  verbose: false

task_pd_gains:
  Kp: [150.0, 150.0, 150.0, 50.0, 50.0, 50.0]
  Kd: [30.0, 30.0, 30.0, 10.0, 11.0, 10.0]
  verbose: false

task_nric_gains:
  k1: [8.0, 8.0, 8.0, 1.0, 1.0, 1.0] # d-gian
  k2: [21.0, 21.0, 21.0, 5.0, 5.0, 5.0] # p-gian
  K: [30.0, 30.0, 30.0, 30.0, 20.0, 20.0, 20.0]
  Kp: [13.0, 13.0, 13.0, 13.0, 8.0, 8.0, 8.0]
  Ki: [4.0, 4.0, 4.0, 4.0, 2.0, 2.0, 2.0]
  gamma: [4.0, 4.0, 4.0, 4.0, 4.0, 4.0, 4.0]
  reflected_inertia: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
  verbose: false

energy_tank_config:
  initial_energy: 0.45 # [J]
  max_energy: 5.0 # [J]
  min_energy: 0.1 # [J]
  Kd: [30.0, 30.0, 30.0, 30.0, 15.0, 15.0, 15.0] # [N*m*s]
  mpc_run_out_time: 5 # [s]
  verbose: false

joint_fric_gains:
  Kp: [50.0, 50.0, 50.0, 50.0, 30.0, 30.0, 30.0]
  Kd: [3.0, 3.0, 3.0, 3.0, 2.0, 2.0, 2.0]
  # Kp: [500.0, 500.0, 500.0, 500.0, 300.0, 300.0, 300.0]
  # Kd: [30.0, 30.0, 30.0, 30.0, 20.0, 20.0, 20.0]
  joint_stiffness_matrix : [8000, 8000, 8000, 8000, 7000, 7000, 7000]
  torque_interface: "COULOMB_OBSERVER" # FIRST_ORDER_FRIC | SECOND_ORDER_FRIC | THIRD_ORDER_FRIC | INERTIA_RESHAPING | L1_FRIC | IMPLICIT_L1_FRIC | COULOMB_OBSERVER
  verbose: false

task_fric_gains:
  Kp: [1000.0, 500.0, 500.0, 5.0, 5.0, 5.0]
  Kd: [80.0, 80.0, 80.0, 2.0, 2.0, 2.0]
  joint_stiffness_matrix : [8000, 8000, 8000, 8000, 7000, 7000, 7000]
  torque_interface: "COULOMB_OBSERVER" # FIRST_ORDER_FRIC | SECOND_ORDER_FRIC | THIRD_ORDER_FRIC | INERTIA_RESHAPING | L1_FRIC | IMPLICIT_L1_FRIC | COULOMB_OBSERVER
  verbose: false

# For torque control interface
first_order_fric_gain:
  L : [0, 0, 0, 0, 0, 0, 0]
  motor_inertia_matrix : [0.4, 0.4, 0.4, 0.4, 0.2, 0.2, 0.2]
  verbose: false

second_order_fric_gain:
  # L : [70, 70, 70, 70, 60, 60, 60] # joint space
  # Lp : [30, 30, 30, 30, 20, 20, 20]
  L : [90, 90, 90, 90, 80, 80, 80] # task space
  Lp : [30, 30, 30, 30, 20, 20, 20]
  motor_inertia_matrix : [0.4, 0.4, 0.4, 0.4, 0.2, 0.2, 0.2]
  verbose: false
  
third_order_fric_gain:
  L : [100, 100, 100, 100, 100, 100, 100]
  Lp : [40, 40, 40, 40, 40, 40, 40]
  Li: [40, 40, 40, 40, 40, 40, 40]
  motor_inertia_matrix : [0.4, 0.4, 0.4, 0.4, 0.2, 0.2, 0.2]
  verbose: false

inertia_reshaping_gain:
  motor_inertia_matrix : [0.4, 0.4, 0.4, 0.4, 0.2, 0.2, 0.2]
  desired_motor_inertia: [0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05]
  verbose: false

l1_fric_gain:
  As : [-5, -5, -5, -5, -5, -5, -5]
  Gamma : [1000, 1000, 1000, 1000, 1000, 1000, 1000]
  W : [100, 100, 100, 100, 80, 80, 80]
  motor_inertia_matrix : [0.4, 0.4, 0.4, 0.4, 0.2, 0.2, 0.2]
  verbose: false

implicit_l1_fric_gain:
  Gamma : [5000, 5000, 5000, 5000, 5000, 5000, 5000]
  Gamma_p : [500, 500, 500, 500, 500, 500, 500]
  W : [120, 120, 120, 120, 100, 100, 100]
  motor_inertia_matrix : [0.4, 0.4, 0.4, 0.4, 0.2, 0.2, 0.2]
  verbose: false  

coulomb_observer_gain:
  K : [10, 10, 10, 10, 10, 10, 10]
  L : [1, 1, 1, 1, 1, 1, 1]
  motor_inertia_matrix : [0.4, 0.4, 0.4, 0.4, 0.2, 0.2, 0.2]
  verbose: false  